xds/src/main/java/io/grpc/xds/EvaluateArgs.java

```diff
@@ -0,0 +1,61 @@
+/*
+ * Copyright 2020 The gRPC Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.grpc.xds;
+
+import com.google.common.collect.ImmutableMap;
+import javax.annotation.Nullable;
+
+/** This data structure provides request arguments to be evaluated in routing selection or
+ *  authorization/http-filtering, to support the {@link io.grpc.xds.Matcher} logic. */
+public abstract class EvaluateArgs {
```

Comment:
This represents some pure data, so I don't see the reason to make it an abstract class. It could just be a final class with absent values returning null.